### PR TITLE
Add eslint-plugin-eslint-plugin and enable/autofix most rules

### PR DESCRIPTION
**Pull request by bmish**
The rules from this plugin enforce a lot of great best practices and should save us a lot of time during code reviews!

https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin

### CODE REVIEW

File: lib/rules/require-computed-property-dependencies.js [hunk 1]
```diff
@@ -412,9 +412,8 @@ module.exports = {
           if (undeclaredKeys.length > 0) {
             context.report({
               node,
-              message: `Use of undeclared dependencies in computed property: ${undeclaredKeys.join(
-                ', '
-              )}`,
+              message: `Use of undeclared dependencies in computed property: {{undeclaredKeys}}`,
+              data: { undeclaredKeys: undeclaredKeys.join(', ') },
```
Comment (Turbo87) [inline lib/rules/require-computed-property-dependencies.js]: what is the advantage of this? 🤔

File: tests/lib/rules/no-old-shims.js [hunk 2]
```diff
@@ -56,7 +56,7 @@ eslintTester.run('no-old-shims', rule, {
     },
     {
       code: "import Sortable from 'ember-controllers/sortable';",
-      output: "import Sortable from 'ember-controllers/sortable';",
+      output: "import Sortable from 'ember-controllers/sortable';", // eslint-disable-line eslint-plugin/prefer-output-null
```
Comment (Turbo87) [inline tests/lib/rules/no-old-shims.js]: why not `null` in this case?

File: tests/lib/rules/no-unnecessary-service-injection-argument.js [hunk 3]
```diff
@@ -32,8 +32,6 @@ ruleTester.run('no-unnecessary-service-injection-argument', rule, {
     "const controller = Controller.extend({ serviceName: service('service-name') });",
 
     // Property name does not match service name:
-    "export default Component.extend({ specialName: service('service-name') });",
-    "export default Component.extend({ specialName: inject('service-name') });",
```
Comment (Turbo87) [inline tests/lib/rules/no-unnecessary-service-injection-argument.js]: were these duplicates?

File: tests/lib/rules/no-old-shims.js (hunk 2 above)
Comment (bmish) [reply tests/lib/rules/no-old-shims.js]: This rule seems to erroneously report that there's an autofix available even though the output is the same as the input. So it doesn't allow me to use `null` here.

File: lib/rules/require-computed-property-dependencies.js (hunk 1 above)
Comment (bmish) [reply lib/rules/require-computed-property-dependencies.js]: This rule prefers placeholders!

https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/master/docs/rules/prefer-placeholders.md

File: tests/lib/rules/no-unnecessary-service-injection-argument.js (hunk 3 above)
Comment (bmish) [reply tests/lib/rules/no-unnecessary-service-injection-argument.js]: Correct!

https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/master/docs/rules/no-identical-tests.md

**Pull request merged**